Flaky-test: AuditorReplicasCheckTest

AuditorReplicasCheckTest has the following tests that are flaky, detected using [Nondex](https://github.com/TestingResearchIllinois/NonDex).
testReplicasCheckForLedgersFoundHavingLessThanAQReplicasOfAnEntry()
testReplicasCheckForLedgersFoundHavingLessThanWQReplicasOfAnEntry()
testReplicasCheckForLedgersFoundHavingNoReplica()
testReplicasCheckForLedgersWithEmptySegments()

All fail with a similar error as below.

```
[INFO] Running org.apache.bookkeeper.replication.AuditorReplicasCheckTest
[ERROR] Tests run: 5, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 8.48 s <<< FAILURE! - in org.apache.bookkeeper.replication.AuditorReplicasCheckTest
[ERROR] org.apache.bookkeeper.replication.AuditorReplicasCheckTest.testReplicasCheckForBookieHandleNotAvailable  Time elapsed: 1.287 s  <<< ERROR!
java.lang.IllegalArgumentException: New entry must have a first entry greater than any existing ensemble key
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:145)
	at org.apache.bookkeeper.client.LedgerMetadataBuilder.newEnsembleEntry(LedgerMetadataBuilder.java:146)
	at org.apache.bookkeeper.replication.AuditorReplicasCheckTest.createClosedLedgerMetadata(AuditorReplicasCheckTest.java:202)
	at org.apache.bookkeeper.replication.AuditorReplicasCheckTest.testReplicasCheckForBookieHandleNotAvailable(AuditorReplicasCheckTest.java:324)

```

--- a/bookkeeper-server/src/test/java/org/apache/bookkeeper/replication/AuditorReplicasCheckTest.java
+++ b/bookkeeper-server/src/test/java/org/apache/bookkeeper/replication/AuditorReplicasCheckTest.java
@@ -28,7 +28,7 @@ import java.net.URI;
 import java.net.UnknownHostException;
 import java.util.ArrayList;
 import java.util.Collections;
-import java.util.HashMap;
+import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.CompletableFuture;
@@ -301,7 +301,7 @@ public class AuditorReplicasCheckTest extends BookKeeperClusterTestCase {
          * BookieHandleNotAvailableException so asyncGetListOfEntriesOfLedger will
          * return BookieHandleNotAvailableException.
          */
-        Map<Long, List<BookieId>> segmentEnsembles = new HashMap<Long, List<BookieId>>();
+        Map<Long, List<BookieId>> segmentEnsembles = new LinkedHashMap<Long, List<BookieId>>();
         segmentEnsembles.put(0L, bookieAddresses);
         long ledgerId = 1L;
         createClosedLedgerMetadata(lm, ledgerId, ensembleSize, writeQuorumSize, ackQuorumSize, segmentEnsembles,
@@ -399,7 +399,7 @@ public class AuditorReplicasCheckTest extends BookKeeperClusterTestCase {
          * Empty one for all of the bookies, so this ledger would be counted in
          * ledgersFoundHavingNoReplicaOfAnEntry .
          */
-        Map<Long, List<BookieId>> segmentEnsembles = new HashMap<Long, List<BookieId>>();
+        Map<Long, List<BookieId>> segmentEnsembles = new LinkedHashMap<Long, List<BookieId>>();
         segmentEnsembles.put(0L, bookieAddresses);
         long ledgerId = 1L;
         createClosedLedgerMetadata(lm, ledgerId, ensembleSize, writeQuorumSize, ackQuorumSize, segmentEnsembles,
@@ -529,7 +529,7 @@ public class AuditorReplicasCheckTest extends BookKeeperClusterTestCase {
          * would be counted towards
          * ledgersFoundHavingLessThanAQReplicasOfAnEntry.
          */
-        Map<Long, List<BookieId>> segmentEnsembles = new HashMap<Long, List<BookieId>>();
+        Map<Long, List<BookieId>> segmentEnsembles = new LinkedHashMap<Long, List<BookieId>>();
         int ensembleSize = 4;
         int writeQuorumSize = 3;
         int ackQuorumSize = 2;
@@ -682,7 +682,7 @@ public class AuditorReplicasCheckTest extends BookKeeperClusterTestCase {
          * for this ledger a copy of entry 3, so this ledger would be counted
          * towards ledgersFoundHavingLessThanWQReplicasOfAnEntry.
          */
-        Map<Long, List<BookieId>> segmentEnsembles = new HashMap<Long, List<BookieId>>();
+        Map<Long, List<BookieId>> segmentEnsembles = new LinkedHashMap<Long, List<BookieId>>();
         int ensembleSize = 4;
         int writeQuorumSize = 3;
         int ackQuorumSize = 2;
@@ -837,7 +837,7 @@ public class AuditorReplicasCheckTest extends BookKeeperClusterTestCase {
          * numLedgersFoundHavingNoReplicaOfAnEntry/LessThanAQReplicasOfAnEntry
          * /WQReplicasOfAnEntry.
          */
-        Map<Long, List<BookieId>> segmentEnsembles = new HashMap<Long, List<BookieId>>();
+        Map<Long, List<BookieId>> segmentEnsembles = new LinkedHashMap<Long, List<BookieId>>();
         int ensembleSize = 4;
         int writeQuorumSize = 3;
         int ackQuorumSize = 2;

